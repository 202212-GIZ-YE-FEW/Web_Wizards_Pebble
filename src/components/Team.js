import Image from "next/image";
import data from "public/data.json";

const Team = () => {
    const teamCards = data.map((member) => {
        const socialLinks = member.links.map((link) => {
            return (
                <a key={link.name} className='icon' href={link.link}>
                    <i
                        className={`fa-brands fa-${link.name}  fa-2xs text-2xl m-2`}
                        aria-hidden='true'
                        title="Adeeb'a GitHub Profile"
                    ></i>
                </a>
            );
        });
        return (
            <div
                key={member.id}
                className='
          w-1/2
          lg:w-1/3
          xl:w-1/4
          px-4'
            >
                <div
                    className='
            m-8
            p-8
            rounded-3xl
            bg-[#29C5E6]
            lg:bg-[#bdd6d0]'
                >
                    <div
                        className='
              relative
              w-[170px]h-170px]
              rounded-full
              z-10
              mx-auto
              mb-6'
                    >
                        <Image
                            src={`/images/${member.image}`}
                            alt={member.name}
                            width={70}
                            height={70}
                            className='rounded-full relative mx-auto'
                        />
                    </div>
                    <div className='text-center'>
                        <h4
                            className='font-medium
            text-lg
            text-dark
            mb-2'
                        >
                            {member.name}
                        </h4>
                    </div>
                    <div className='flex justify-center'>{socialLinks}</div>
                </div>
            </div>
        );
    });
    return (
        <section
            id='team'
            className='
        p-10
        lg:pt-[120px]
        pb-10
        lg:pb-20'
        >
            <div className='container'>
                <div
                    className='
          flex
          flex-wrap
          -mx-4'
                >
                    <div className='w-full px-4'>
                        <div className='text-center mx-auto mb-[60px] max-w-[620px]'>
                            <h2
                                className='
                  font-bold
                  text-3xl
                  sm:text-4xl
                  md:text-[42px]
                  text-dark
                  mb-4
                '
                            >
                                Our Team
                            </h2>
                        </div>
                    </div>
                </div>
                <div className='flex flex-wrap justify-center -mx-4'>
                    {teamCards}
                </div>
            </div>
        </section>
    );
};

export default Team;
