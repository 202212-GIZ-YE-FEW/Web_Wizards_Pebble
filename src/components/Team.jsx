import Image from "next/image";

import teamInfo from "~/teamInfo.json";

const Team = (props) => {
    const { t } = props;

    const teamCard = teamInfo.map((member) => {
        const teamSocial = member.links.map((link) => {
            return (
                <a key={link.name} href={link.link}>
                    <i
                        className={`fa-brands fa-${link.name} text-white text-lv m-2`}
                        aria-hidden='true'
                        title={`${member.name}'s ${link.name} Profile`}
                    ></i>
                </a>
            );
        });
        return (
            <div
                key={member.id}
                className='w-full sm:w-1/2 lg:w-1/3 xl:w-1/4 px-4'
            >
                <div className='m-8 py-6 rounded-3xl bg-[#29C5E6] lg:bg-[#bdd6d0]'>
                    <div
                        className='
          relative w-20 h-20 rounded-full z-10 mx-auto'
                    >
                        <Image
                            src={`/images/${member.image}`}
                            alt={member.name}
                            width={100}
                            height={100}
                            className='w-[100%] h-[100%] mx-auto rounded-full'
                        />
                    </div>
                    <div className='text-center'>
                        <h4 className='pt-2 pb-2 text-center text-[10px] font-medium md:pb-2 md:pt-2 md:text-[24px]'>
                            {member.name}
                        </h4>
                    </div>
                    <div className='flex justify-center'>{teamSocial}</div>
                </div>
            </div>
        );
    });
    return (
        <section
            id='team'
            className='m-10 min-h-[320px] overflow-hidden rounded-3xl bg-[#fef1e6] md:m-20 md:min-h-[900px] lg:m-44'
        >
            <h2 className='p-4 text-center text-2xl font-medium md:p-8 md:text-4xl md:font-bold lg:p-12 lg:text-5xl lg:font-bold'>
                {t("ourTeam")}
            </h2>
            <div className='flex flex-wrap justify-center'>{teamCard}</div>
        </section>
    );
};

export default Team;
