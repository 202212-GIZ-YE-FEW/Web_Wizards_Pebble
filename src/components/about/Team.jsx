import Image from "next/image";

import teamInfo from "~/teamInfo.json";

const Team = (props) => {
    const { t } = props;

    const teamCard = teamInfo.map((member) => {
        const teamSocial = member.links.map((link) => {
            return (
                <a key={link.name} href={link.link}>
                    <i
                        className={`fa-brands fa-${link.name} text-white text-lg md:text-2xl mx-2`}
                        aria-hidden='true'
                        title={`${member.name}'s ${link.name} Profile`}
                    ></i>
                </a>
            );
        });
        return (
            <div
                key={member.id}
                className='m-2 h-28 w-28 rounded-2xl lg:bg-secondary-100 bg-[#29C5E6] md:m-4 md:h-56 md:w-56 lg:m-8 lg:h-72 lg:w-72'
            >
                <Image
                    src={`/Images/team/${member.image}`}
                    alt={member.name}
                    width={100}
                    height={100}
                    className='w-10 h-10 md:w-20 md:h-20 md:my-4 lg:w-32 lg:h-32 lg:my-6 mx-auto rounded-full mt-2'
                />
                <h4 className='pt-2 pb-2 text-center text-[10px] font-medium md:pb-2 md:pt-2 md:text-[24px]'>
                    {member.name}
                </h4>
                <div className='flex justify-center'>{teamSocial}</div>
            </div>
        );
    });
    return (
        <section
            id='team'
            className='m-4 p-5 min-h-[320px] rounded-3xl bg-[#fef1e6] md:m-20 md:min-h-[900px] lg:m-8'
        >
            <h5 className='text-center mb-4'>{t("ourTeam")}</h5>
            <h2 className='p-4 text-center text-2xl md:p-8 md:text-4xl font-bold lg:p-12 lg:text-5xl lg:font-bold m-1 md:m-4 font-sans'>
                Web Wizard
            </h2>
            <div className='flex flex-wrap justify-around'>{teamCard}</div>
        </section>
    );
};

export default Team;
