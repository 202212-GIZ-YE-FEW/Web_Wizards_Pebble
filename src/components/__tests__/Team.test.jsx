import renderer from "react-test-renderer";

import Team from "../about/Team";

it("renders correctly", () => {
    const tree = renderer.create(<Team t={(str) => str} />).toJSON();
    expect(tree).toMatchSnapshot();
});
