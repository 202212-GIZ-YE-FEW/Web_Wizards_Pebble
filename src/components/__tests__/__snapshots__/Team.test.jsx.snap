// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="m-4 p-5 min-h-[320px] rounded-3xl bg-[#fef1e6] md:m-20 md:min-h-[900px] lg:m-8"
  id="team"
>
  <h5
    className="text-center mb-4"
  >
    ourTeam
  </h5>
  <h2
    className="p-4 text-center text-2xl md:p-8 md:text-4xl font-bold lg:p-12 lg:text-5xl lg:font-bold m-1 md:m-4 font-sans"
  >
    Web Wizard
  </h2>
  <div
    className="flex flex-wrap justify-around"
  >
    <div
      className="m-2 h-28 w-28 rounded-2xl lg:bg-secondary-100 bg-[#29C5E6] md:m-4 md:h-56 md:w-56 lg:m-8 lg:h-72 lg:w-72"
    >
      <img
        alt="Abdullah Binsaed"
        className="w-10 h-10 md:w-20 md:h-20 md:my-4 lg:w-32 lg:h-32 lg:my-6 mx-auto rounded-full mt-2"
        data-nimg="1"
        decoding="async"
        height={100}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Fteam%2Fabdullah.png&w=256&q=75"
        srcSet="/_next/image?url=%2Fimages%2Fteam%2Fabdullah.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fteam%2Fabdullah.png&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={100}
      />
      <h4
        className="pt-2 pb-2 text-center text-[10px] font-medium md:pb-2 md:pt-2 md:text-[24px]"
      >
        Abdullah Binsaed
      </h4>
      <div
        className="flex justify-center"
      >
        <a
          href="https://github.com/iiAbady"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-github text-white text-lg md:text-2xl mx-2"
            title="Abdullah Binsaed's github Profile"
          />
        </a>
        <a
          href="http://"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-discord text-white text-lg md:text-2xl mx-2"
            title="Abdullah Binsaed's discord Profile"
          />
        </a>
        <a
          href="http://"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-linkedin text-white text-lg md:text-2xl mx-2"
            title="Abdullah Binsaed's linkedin Profile"
          />
        </a>
      </div>
    </div>
    <div
      className="m-2 h-28 w-28 rounded-2xl lg:bg-secondary-100 bg-[#29C5E6] md:m-4 md:h-56 md:w-56 lg:m-8 lg:h-72 lg:w-72"
    >
      <img
        alt="Sufyan Basurra"
        className="w-10 h-10 md:w-20 md:h-20 md:my-4 lg:w-32 lg:h-32 lg:my-6 mx-auto rounded-full mt-2"
        data-nimg="1"
        decoding="async"
        height={100}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Fteam%2Fsufyan.png&w=256&q=75"
        srcSet="/_next/image?url=%2Fimages%2Fteam%2Fsufyan.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fteam%2Fsufyan.png&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={100}
      />
      <h4
        className="pt-2 pb-2 text-center text-[10px] font-medium md:pb-2 md:pt-2 md:text-[24px]"
      >
        Sufyan Basurra
      </h4>
      <div
        className="flex justify-center"
      >
        <a
          href="https://github.com/SufyanCS"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-github text-white text-lg md:text-2xl mx-2"
            title="Sufyan Basurra's github Profile"
          />
        </a>
        <a
          href="http://"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-facebook text-white text-lg md:text-2xl mx-2"
            title="Sufyan Basurra's facebook Profile"
          />
        </a>
        <a
          href="http://"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-linkedin text-white text-lg md:text-2xl mx-2"
            title="Sufyan Basurra's linkedin Profile"
          />
        </a>
      </div>
    </div>
    <div
      className="m-2 h-28 w-28 rounded-2xl lg:bg-secondary-100 bg-[#29C5E6] md:m-4 md:h-56 md:w-56 lg:m-8 lg:h-72 lg:w-72"
    >
      <img
        alt="Adeeb Abdulsamad"
        className="w-10 h-10 md:w-20 md:h-20 md:my-4 lg:w-32 lg:h-32 lg:my-6 mx-auto rounded-full mt-2"
        data-nimg="1"
        decoding="async"
        height={100}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Fteam%2Fadeeb.png&w=256&q=75"
        srcSet="/_next/image?url=%2Fimages%2Fteam%2Fadeeb.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fteam%2Fadeeb.png&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={100}
      />
      <h4
        className="pt-2 pb-2 text-center text-[10px] font-medium md:pb-2 md:pt-2 md:text-[24px]"
      >
        Adeeb Abdulsamad
      </h4>
      <div
        className="flex justify-center"
      >
        <a
          href="https://github.com/adeeb-samad"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-github text-white text-lg md:text-2xl mx-2"
            title="Adeeb Abdulsamad's github Profile"
          />
        </a>
        <a
          href="http://"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-instagram text-white text-lg md:text-2xl mx-2"
            title="Adeeb Abdulsamad's instagram Profile"
          />
        </a>
        <a
          href="https://www.linkedin.com/in/adeeb-samad/"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-linkedin text-white text-lg md:text-2xl mx-2"
            title="Adeeb Abdulsamad's linkedin Profile"
          />
        </a>
      </div>
    </div>
    <div
      className="m-2 h-28 w-28 rounded-2xl lg:bg-secondary-100 bg-[#29C5E6] md:m-4 md:h-56 md:w-56 lg:m-8 lg:h-72 lg:w-72"
    >
      <img
        alt="Azzam Al-Sulmi"
        className="w-10 h-10 md:w-20 md:h-20 md:my-4 lg:w-32 lg:h-32 lg:my-6 mx-auto rounded-full mt-2"
        data-nimg="1"
        decoding="async"
        height={100}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Fteam%2Fazzam.png&w=256&q=75"
        srcSet="/_next/image?url=%2Fimages%2Fteam%2Fazzam.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fteam%2Fazzam.png&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={100}
      />
      <h4
        className="pt-2 pb-2 text-center text-[10px] font-medium md:pb-2 md:pt-2 md:text-[24px]"
      >
        Azzam Al-Sulmi
      </h4>
      <div
        className="flex justify-center"
      >
        <a
          href="http://github.com/sulmi24"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-github text-white text-lg md:text-2xl mx-2"
            title="Azzam Al-Sulmi's github Profile"
          />
        </a>
        <a
          href="http://"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-twitter text-white text-lg md:text-2xl mx-2"
            title="Azzam Al-Sulmi's twitter Profile"
          />
        </a>
        <a
          href="http://linkedin.com/in/a-sulmi24"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-linkedin text-white text-lg md:text-2xl mx-2"
            title="Azzam Al-Sulmi's linkedin Profile"
          />
        </a>
      </div>
    </div>
    <div
      className="m-2 h-28 w-28 rounded-2xl lg:bg-secondary-100 bg-[#29C5E6] md:m-4 md:h-56 md:w-56 lg:m-8 lg:h-72 lg:w-72"
    >
      <img
        alt="Bakeel Hamad"
        className="w-10 h-10 md:w-20 md:h-20 md:my-4 lg:w-32 lg:h-32 lg:my-6 mx-auto rounded-full mt-2"
        data-nimg="1"
        decoding="async"
        height={100}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Fteam%2Fbakeel.png&w=256&q=75"
        srcSet="/_next/image?url=%2Fimages%2Fteam%2Fbakeel.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fteam%2Fbakeel.png&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={100}
      />
      <h4
        className="pt-2 pb-2 text-center text-[10px] font-medium md:pb-2 md:pt-2 md:text-[24px]"
      >
        Bakeel Hamad
      </h4>
      <div
        className="flex justify-center"
      >
        <a
          href="https://github.com/HamadBakeel/HamadBakeel"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-github text-white text-lg md:text-2xl mx-2"
            title="Bakeel Hamad's github Profile"
          />
        </a>
        <a
          href="http://"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-discord text-white text-lg md:text-2xl mx-2"
            title="Bakeel Hamad's discord Profile"
          />
        </a>
        <a
          href="http://"
        >
          <i
            aria-hidden="true"
            className="fa-brands fa-linkedin text-white text-lg md:text-2xl mx-2"
            title="Bakeel Hamad's linkedin Profile"
          />
        </a>
      </div>
    </div>
  </div>
</section>
`;
